packages/datastore/src/datastore.ts

```diff
@@ -92,27 +131,91 @@ class Datastore implements IIterable<Table> {
    * `O(log32 n)`
    */
   get<S extends Schema>(schema: S): Table<S> {
-    throw '';
+    return this._tables[schema.id] as Table<S>;
   }
 
   /**
+   * Begin a new transaction in the store.
+   *
+   * @returns The id of the new transaction
    *
+   * #### Notes
+   * This will allow the state of the store to be mutated
+   * thorugh the `update` method on the individual tables.
+   *
+   * After the updates are completed, `endTransaction` should
+   * be called.
    */
   beginTransaction(): string {
-    throw '';
+    const id = createDuplexId(this.version, this.id);
+    this._initTransaction(id);
+    return id;
   }
 
   /**
+   * Completes a transaction.
    *
+   * #### Notes
+   * This completes a transaction previously started with
+   * `beginTransaction`. If a change has occurred, the
+   * `changed` signal will be emitted.
    */
   endTransaction(): void {
-    throw '';
+    this._finalizeTransaction();
+    if (this._context.change) {
+      this._changed.emit({
+        storeId: this.id,
+        transactionId: this._currentTransactionId,
+        type: 'transaction',
+        change: this._context.change,
+      });
+    }
+    if (this.transactionBroadcastHandler && this._context.patch) {
+      MessageLoop.sendMessage(
+        this.transactionBroadcastHandler,
+        new Datastore.TransactionMessage({
+          id: this._currentTransactionId,
+          storeId: this.id,
+          patch: this._context.patch,
+      }));
+    }
+  }
+
+
+  /**
+   * Apply a transaction to the datastore.
+   *
+   * @param transaction - The data of the transaction.
+   *
+   * @returns A promise which resolves when the action is complete.
+   *
+   * @throws An exception if `apply` is called during a mutation.
+   *
+   * #### Notes
+   * If changes are made, the `changed` signal will be emitted.
+   */
+  apply(transaction: Datastore.Transaction): void {
```

Comment:
This method name is a bit generic. Should it be public? 

I thought we had decided to make the datastore a message handler as well?